datahub-web-react/src/app/entity/shared/siblingUtils.ts

```diff
@@ -169,35 +170,47 @@ export const shouldEntityBeTreatedAsPrimary = (extractedBaseEntity: { siblings?:
     return isPrimary;
 };
 
-export const combineEntityDataWithSiblings = <T>(baseEntity: T): T => {
-    if (!baseEntity) {
-        return baseEntity;
-    }
-    const baseEntityKey = Object.keys(baseEntity)[0];
-    const extractedBaseEntity = baseEntity[baseEntityKey];
-
+const combineEntityWithSiblings = (entity: GenericEntityProperties) => {
     // eslint-disable-next-line @typescript-eslint/dot-notation
-    const siblingAspect = extractedBaseEntity.siblings;
+    const siblingAspect = entity.siblings;
     if ((siblingAspect?.siblings || []).length === 0) {
-        return baseEntity;
+        return entity;
     }
 
     // eslint-disable-next-line @typescript-eslint/dot-notation
-    const siblings: T[] = siblingAspect?.siblings || [];
+    const siblings = siblingAspect?.siblings || [];
 
-    const isPrimary = shouldEntityBeTreatedAsPrimary(extractedBaseEntity);
+    const isPrimary = shouldEntityBeTreatedAsPrimary(entity);
 
     const combinedBaseEntity: any = siblings.reduce(
         (prev, current) =>
             merge(clean(isPrimary ? current : prev), clean(isPrimary ? prev : current), {
                 arrayMerge: combineMerge,
                 customMerge: customMerge.bind({}, isPrimary),
             }),
-        extractedBaseEntity,
-    ) as T;
+        entity,
+    );
 
     // Force the urn of the combined entity to the current entity urn.
-    combinedBaseEntity.urn = extractedBaseEntity.urn;
+    combinedBaseEntity.urn = entity.urn;
+
+    return combinedBaseEntity;
+};
+
+export const combineEntityDataWithSiblings = <T>(baseEntity: T): T => {
+    if (!baseEntity) {
+        return baseEntity;
+    }
+    const baseEntityKey = Object.keys(baseEntity)[0];
```

Comment:
@chriscollins3456 just seeing this now. What is the idea behind converting the object to keys and taking whatever the first entry might be?